### PR TITLE
Support GPU init in python test

### PR DESCRIPTION
**Pull request by cydrain**

### CODE REVIEW

File: include/knowhere/gpu/gpu_res_mgr.h [hunk 1]
```diff
@@ -117,25 +121,22 @@ class GPUResMgr {
...
     GetRes() {
-        if (init_) {
-            auto res = res_bq_.Take();
-            return res;
-        } else {
-            KNOWHERE_THROW_MSG("GPUResMgr not initialized");
-        }
+        // Generally Init() should be called separately,
+        // here is for supporting python test
+        Init();
```
Comment (Presburger) [inline include/knowhere/gpu/gpu_res_mgr.h:126]: check every time? init func have locks.

Comment (cydrain) [reply include/knowhere/gpu/gpu_res_mgr.h:126]: after initialized, the lock will not be used any more

**Pull request merged**